pkg/server/api/svid/v1/service.go

```diff
@@ -5,68 +5,56 @@ import (
 	"crypto/x509"
 	"time"
 
+	"github.com/sirupsen/logrus"
 	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/spire/pkg/common/telemetry"
 	"github.com/spiffe/spire/pkg/common/x509util"
 	"github.com/spiffe/spire/pkg/server/api"
 	"github.com/spiffe/spire/pkg/server/api/rpccontext"
 	"github.com/spiffe/spire/pkg/server/ca"
+	"github.com/spiffe/spire/pkg/server/plugin/datastore"
+	"github.com/spiffe/spire/pkg/server/util/regentryutil"
+	"github.com/spiffe/spire/proto/spire/common"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
 // Config is the service configuration
 type Config struct {
 	ServerCA    ca.ServerCA
+	DataStore   datastore.DataStore
```

Comment:
I wonder if this wouldn't be easier to test if the service took a simple interface here instead of being coupled to the datastore.

```
type AuthorizedEntryFetcher interface {
    FetchAuthorizedEntries(ctx context.Context, agentID spiffeid.ID) ([]*types.Entry, error)
}
```